spec/core/query_spec.rb

```diff
@@ -267,6 +267,12 @@
           ).to eq(answer)
       end
 
+      let(:dv1) { Daru::Vector.new([1,11,32,Float::NAN,nil]) }
+      let(:dv2) { Daru::Vector.new([1,11]) }
+      it "handles empty data" do
+        expect(dv1.where(dv1.lt(14))).to eq(dv2)
```

Comment:
I am not sure you need to create separate `dv`, `expect(dv1.where(dv1.lt(14))).to eq(Daru::Vector.new([1,11]))` should be enough.